src/Handler/IdInjection/RequestIdInjector.php

```diff
@@ -0,0 +1,22 @@
+<?php
+declare(strict_types=1);
+
+namespace Chimera\Routing\Handler\IdInjection;
+
+use Psr\Http\Message\ServerRequestInterface;
+use Ramsey\Uuid\UuidInterface;
+
+interface RequestIdInjector
```

Comment:
As I mentioned above, I believe this seems a bit too complicated for the problem we're trying to solve here. IMHO introducing this abstraction sounds a bit over-engineering.